OpenRA.Mods.Common/Warheads/DestroyResourceWarhead.cs

```diff
@@ -39,9 +52,44 @@ public override void DoImpact(Target target, WarheadArgs args)
 			var minRange = (Size.Length > 1 && Size[1] > 0) ? Size[1] : 0;
 			var allCells = world.Map.FindTilesInAnnulus(targetTile, minRange, Size[0]);
 
-			// Destroy all resources in the selected tiles
+			var removeAllTypes = ResourceTypes.Count == 0;
+
+			var averageRemoval = ResourceAmount * 1000 / allCells.Count();
+			var accumulatedRemoval = 0;
+			var currentResourceAmount = 0;
+
 			foreach (var cell in allCells)
-				resLayer.Destroy(cell);
+			{
+				var cellResourceType = resLayer.GetResourceType(cell);
+				if (cellResourceType == null)
+					continue;
+
+				if (removeAllTypes || ResourceTypes.Contains(cellResourceType.Info.Type))
+				{
+					if (ResourceAmount <= 0)
+						resLayer.Destroy(cell);
+					else if (ResourceRemovalType == ResourceRemoval.PerTile)
+						resLayer.Harvest(cell, ResourceAmount);
```

Comment:
We should rename this, calling `Harvest` here sounds odd.

Reply:
The problem is that this is used primarily for harvesters and returns the resource type of the harvested cell... I'm looking for names right now, but I can't find a fitting one. What about `Take` or `Remove`?

Reply:
`Take` sounds good to me.

Reply:
Renamed.